src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -1,47 +1,37 @@
 ﻿using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
+using Dynamo.Properties;
 using Dynamo.Search.SearchElements;
 using Dynamo.Wpf.ViewModels;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.Mirror;
+using ProtoCore.Utils;
 
 namespace Dynamo.ViewModels
 {
     /// <summary>
-    /// Search View Model for Node AutoComplate Search Bar
+    /// Search View Model for Node AutoComplete Search Bar
     /// </summary>
     public class NodeAutoCompleteSearchViewModel : SearchViewModel
     {
+
         internal PortViewModel PortViewModel { get; set; }
 
         internal NodeAutoCompleteSearchViewModel(DynamoViewModel dynamoViewModel) : base(dynamoViewModel)
         {
             // Do nothing for now, but we may off load some time consuming operation here later
         }
 
-        internal void InitializeDefaultAutoCompleteCandidates()
-        {
-            var candidates = new List<NodeSearchElementViewModel>();
-            // TODO: These are hard copied all time top 7 nodes placed by customers
-            // This should be only served as a temporary default case.
-            var queries = new List<string>(){ "Code Block", "Watch", "List Flatten", "List Create", "String", "Double", "Python" };
-            foreach (var query in queries)
-            {
-                var foundNode = Search(query).ToList().FirstOrDefault();
-                if(foundNode != null)
-                {
-                    candidates.Add(foundNode);
-                }
-            }
-            FilteredResults = candidates;
-        }
-
         internal void PopulateAutoCompleteCandidates()
         {
-            if(PortViewModel == null) return;
+            if (PortViewModel == null) return;
 
             var searchElements = GetMatchingNodes();
             FilteredResults = searchElements.Select(e =>
             {
-                var vm  = new NodeSearchElementViewModel(e, this);
+                var vm = new NodeSearchElementViewModel(e, this);
+                //TODO I think this leaks.
```

Comment:
I think we can remove this TODO as I added the proposed change below to unsubscribe the event.

Reply:
will do.